File: mobile-tests/pages/registration_page.py
# pages/registration_page.py

from appium.webdriver.common.appiumby import AppiumBy
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class RegistrationPage:
    def __init__(self, driver):
        self.driver = driver

    def navigate(self, base_url):
        self.driver.get(base_url)
        WebDriverWait(self.driver, 5).until(
            EC.element_to_be_clickable((AppiumBy.XPATH, '//a[normalize-space()="Sign up"]'))
        ).click()

    def register(self, username, email, password):
        self.driver.find_element(AppiumBy.XPATH, '//input[@placeholder="Username"]').send_keys(username)
        self.driver.find_element(AppiumBy.XPATH, '//input[@placeholder="Email"]').send_keys(email)
        self.driver.find_element(AppiumBy.XPATH, '//input[@placeholder="Password"]').send_keys(password)

        WebDriverWait(self.driver, 5).until(
            EC.element_to_be_clickable((AppiumBy.XPATH, '//button[normalize-space()="Sign up"]'))
        ).click()

        WebDriverWait(self.driver, 15).until(
            EC.url_contains('/')
        )

    def get_error_messages(self):
        error_elements = self.driver.find_elements(AppiumBy.CSS_SELECTOR, '.error-messages li')
        return [el.text for el in error_elements]

File: ui-tests/pages/RegistrationPage.ts
import { Page } from '@playwright/test';

export class RegistrationPage {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async navigate() {
    await this.page.goto('/');
    await this.page.waitForSelector('text=Sign up', { timeout: 3000 });
    await this.page.click('text=Sign up');
    await this.page.waitForSelector('input[placeholder="Username"]', { timeout: 3000 });
  }

  async register(username: string, email: string, password: string) {
  await this.page.fill('input[placeholder="Username"]', username);
  await this.page.fill('input[placeholder="Email"]', email);
  await this.page.fill('input[placeholder="Password"]', password);
  await this.page.waitForSelector('button:has-text("Sign up")', { timeout: 3000 });

  await Promise.all([
    this.page.waitForURL('**/', { timeout: 15000 }),
    this.page.click('button:has-text("Sign up")'),
  ]);
}

  async getErrorMessages(): Promise<string[]> {
    return this.page.locator('.error-messages li').allTextContents();
  }
}
